pi-blaster.c

```diff
@@ -868,7 +880,12 @@ go_go_go(void)
 				//fprintf(stderr, "Bad input: %s", lineptr);
 				n = sscanf(lineptr, "release %d", &servo);
 				if (n != 1 || nl != '\n') {
-					fprintf(stderr, "Bad input: %s", lineptr);
+					n = sscanf(lineptr, "*=%f", &value);
```

Comment:
We really need to simplify this function. It's getting hard to read ;)

Maybe we could move this to line 877, after the two first test and do:

```c
  if (!strcmp(lineptr, "debug_regs\n")) {
    debug_dump_hw();
  } else if (!strcmp(lineptr, "debug_samples\n")) {
    debug_dump_samples();
  } else if (!strcmp(lineptr, "release %d%c", &servo, &nl) == 2 && nl == '\n') {
    release_pwm(servo);
  } else if (sscanf(lineptr, "*=%f%c", &value, &nl) == 2 && nl == '\n') {
    set_all_pwm(value);
  } else if (!strcmp(lineptr, "%d=%f%c", &servo, &value, &nl) == 3 && nl == '\n') {
    if (servo < 0) {
      fprintf(stderr, "Invalid channel number %d\n", servo);
    } else if (value < 0 || value > 1) {
      fprintf(stderr, "Invalid value %f\n", value);
    } else {
      set_pwm(servo, value);
    }
  }
  else {
    fprintf(stderr, "Bad input: %s", lineptr);
  }
```

I think this would make things easier to read. What do you think?

This will need to be tested and you should also add a validation to make sure that `value` is between 0 and 1. This is missing in your code.